### PR TITLE
Set Container Resources In YAML Manifests

### PR DESCRIPTION
**Pull request by seanmalloy**
Prior to this commit the YAML manifests used to install the descheduler
Job and CronJob did not set container requests or limits. This is
considered an anti-pattern when deploying applications on k8s.

Set descheduler container resources to make it a burstable pod. This
will ensure a high quality experience for end users when deploying
descheduler into their clusters using the instructions from the README.

The values choosen for CPU/Memory are not based on any real data.

### CODE REVIEW

**Comment by seanmalloy** [comment]
This fixes the first part of #432. After this is merged I'll also update the helm chart to do the same thing.

/assign @damemi 
/cc @ingvagabund

**Comment by ingvagabund** [comment]
Do we know if the resource constraints are sufficient enough/not sufficient?

**Comment by damemi** [comment]
> Do we know if the resource constraints are sufficient enough/not sufficient?

Yeah, also curious about these numbers (especially the limits) and if we have any way to measure them.

**Comment by seanmalloy** [comment]
All I did was test it on the kind cluster on my laptop, so this really doesn't say much. :-)

I believe there should be a way to allow people to override the requests/limits using kustomize. So I could set some sane defaults and then allow people to override it. Would this approach work for you?

In the short term I don't have time to extensively test the requests/limits a real world environment. Hopefully in the future I can do this, but not right now.

**Comment by seanmalloy** [comment]
Or an even simpler solution would be to only set the requests and not set the limits. Let me know what you think.

**Comment by damemi** [comment]
@seanmalloy  I was wondering if just setting the requests would be sufficient, I think for that we can definitely just go with reasonable values

**Comment by seanmalloy** [comment]
@damemi and @ingvagabund I amended the commit to only sets request.

**Review by damemi — APPROVED** [review]
/approve
thanks @seanmalloy

**Comment by seanmalloy** [comment]
Just need a LGTM from someone.

@ingvagabund or @lixiang233 please review when you have some time. Thanks!

/cc @lixiang233 @ingvagabund

**Pull request merged**
